src/expertsystem/reaction/conservation_rules.py

```diff
@@ -422,8 +424,14 @@ def _check_particles_identical(
     return True
 
 
+@attr.s(frozen=True)
```

Comment:
I guess in preparation of #369?

Reply:
Yes. I my view, the reaction/solving part has to purely work on individual quantum numbers and never create any composite quantum number data structures with logic inside them. The logic should be handled by the rules.
Note that this is a rather subtle but yet largely effecting change. Once spin validation rule verifies the correctness of a spin, it does not need to be checked over and over again. That previously happened, since the regular Spin data structure was used. Alternatively one could also just pass the individual quantum numbers around, but its more convenient to work with this composite Spin data structure.